server/src/main/java/org/opensearch/gateway/GatewayMetaState.java

```diff
@@ -123,9 +124,13 @@ public void start(
         MetaStateService metaStateService,
         MetadataIndexUpgradeService metadataIndexUpgradeService,
         MetadataUpgrader metadataUpgrader,
-        PersistedClusterStateService persistedClusterStateService
+        PersistedClusterStateService persistedClusterStateService,
+        RemoteClusterStateService remoteClusterStateService,
+        PersistedStateRegistry persistedStateRegistry
     ) {
-        assert persistedState.get() == null : "should only start once, but already have " + persistedState.get();
+        this.persistedStateRegistry = persistedStateRegistry;
+        assert persistedStateRegistry.getPersistedState(PersistedStateType.LOCAL) == null : "should only start once, but already have "
```

Comment:
you can assert to ensure registry is set once as you are asserting for individual PersistedState inside the registry class